src/scoped.component.js

```diff
@@ -127,6 +151,48 @@ export class Scoped extends React.Component {
       delete styleTags[this.props.css];
     }
   }
+
+  getPostcssSelector = (id) => {
+    return `style_${id.slice(1)}`;
+  }
+
+  newPostcssState = (props) => {
+    // check if we need to add the style to the head
+    let styleRef = document.head.querySelector(`.${this.getPostcssSelector(this.props.postcss.id)}`);
+    // there's no style - create a new one and add it to the head
+    if (!styleRef) {
+      const style = document.createElement('style');
+      style.setAttribute('class', this.getPostcssSelector(this.props.postcss.id));
+      style.setAttribute('type', 'text/css');
+      style.innerHTML = props.postcss.styles;
+      style.counter = 1;
+      document.head.appendChild(style);
+      styleRef = style;
+    }
+    // there is a style - update the counter property
+    else {
+      styleRef.counter = styleRef.counter + 1;
+    }
+    return { styleRef }
+  }
+
+  updatePostcssStyle = (oldPostcss, newPostcss) => {
+    // check if another component already updated it
+    if (!this.state.styleRef.classList.contains(this.getPostcssSelector(newPostcss.id))) {
+      this.state.styleRef.classList.replace(
+        this.getPostcssSelector(oldPostcss.id),
+        this.getPostcssSelector(newPostcss.id)
+      );
+      this.state.styleRef.innerHTML = newPostcss.styles;  
```

Comment:
i think it might be possible here for some weird stuff to happen. If two components pass in the same post css id and styles but then one of them changes the post css id, shouldn't we look for a new `styleRef` instead of reusing the one we have. So that the other component doesn't accidentally get the new styles?

I might be getting confused here, but wanted to ask.